account_invoice_comment_template/tests/test_account_invoice_report.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 Simone Rubino - Agile Business Group
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountInvoiceReport(TransactionCase):
+    at_install = False
+    post_install = True
+
+    def setUp(self):
+        super().setUp()
+        self.base_comment_model = self.env["base.comment.template"]
+        self.move_obj = self.env["ir.model"].search([("model", "=", "account.move")])
+        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
+        self.before_comment = self._create_comment("before_lines")
+        self.after_comment = self._create_comment("after_lines")
+        self.invoice = self.env["account.move"].create(
+            {
+                "partner_id": self.partner.id,
+                "comment_template1_id": self.before_comment.id,
+                "comment_template2_id": self.after_comment.id,
+            }
+        )
+        self.invoice._set_note1()
+        self.invoice._set_note2()
+
+    def _create_comment(self, position):
+        return self.base_comment_model.create(
+            {
+                "name": "Comment " + position,
+                "position": position,
+                "text": "Text " + position,
+                "partner_ids": [(6, 0, self.partner.ids)],
+                "model_ids": [(6, 0, self.move_obj.ids)],
+            }
+        )
+
+    def test_comments_in_invoice(self):
+        self.invoice._onchange_partner_id()
+        res = (
+            self.env["ir.actions.report"]
+            ._get_report_from_name("account.report_invoice")
+            .render_qweb_html(self.invoice.ids)
+        )
+        self.assertRegexpMatches(str(res[0]), self.before_comment.text)
+        self.assertRegexpMatches(str(res[0]), self.after_comment.text)
```

Comment:
I think it's not responsibility of this module to test that a report prints its values correctly. Also, if this module is tested together with another module that hides those texts that are being tested here, then the test may fail.